code/datums/components/suicide_count.dm

```diff
@@ -0,0 +1,60 @@
+// no extra data displayed (exclusively using callback)
+#define SUICIDE_VIS_NONE 0
+// displays suicide data to holyroles (chaplain)
+#define SUICIDE_VIS_HOLY 1
+// displays suicide data to everyone
+#define SUICIDE_VIS_ALL 2
+
+/datum/component/suicide_count
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/view_mode
+	var/last_person
+	var/count = 0
+	var/datum/callback/ondie
+
+/datum/component/suicide_count/Initialize(_view_mode = SUICIDE_VIS_HOLY, datum/callback/_ondie)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	view_mode = _view_mode
+	ondie = _ondie
+
+/datum/component/suicide_count/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_HUMAN_SUICIDE_COMPLETE, .proc/on_suicide)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/suicide_count/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(COMSIG_HUMAN_SUICIDE_COMPLETE, COMSIG_PARENT_EXAMINE))
+
+/datum/component/suicide_count/on_suicide(mob/living/user)
```

Comment:
This isn't the user, it's the source. This code doesn't work.

Reply:
so if i named it mob/living/source, would it wok fine?

Reply:
No because this is registered on the item, this is the source, aka an `/obj/item`.